plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -700,10 +702,15 @@ public void finishInsertTable(ConnectorSession session, JdbcOutputTableHandle ha
             throw new TrinoException(JDBC_ERROR, e);
         }
         finally {
-            dropTable(session, temporaryRemoteTableName);
+            dropTable(session, temporaryTable);
         }
     }
 
+    protected String buildInsertSql(ConnectorSession session, RemoteTableName targetTable, RemoteTableName sourceTable, String columnNames)
```

Comment:
Pass a `List<String> columnNames`

Reply:
Thank you. How did I miss it second time in a row?